### DISCUSSION TITLE
No-navigation mode for LMS use

### DISCUSSION THREAD

**Discussion by Alex-Jordan** [openwebwork/webwork2#1182] [Ideas]
I've been kicking an idea round in my head, and I wanted to put it up here for comment.

Imagine you use `$LTIGradeMode='homework';`. So you are putting external learning tool links in your LMS to specific problem sets in the WW course. It is often undesirable for the student to follow such a link and then start navigating around to other places within WW. If you want them to visit another assignment, you generally want them to do so via some other ELT link.

What if there was another config variable (or another option for `$LTIGradeMode`) that made pages load without any navigation? No breadcrumbs. No nav panel on the left. It would still have navigation for within the problem set or quiz (like "Next Problem"). Instructors could still get all the navigation. Then you could use the ELT link without concern for students straying from that one set. Also it would become more pleasant to open the ELT link in a frame within the LMS, since the side panel would be gone.

Does this sound useful? (To us at our school I think it would go over well). Can you imagine any bad side effects?

**Comment by awmorp** [comment]
Yes, I think this would be a good addition. We would use it at our university. One thing from the left-hand navigation panel that I think it is important to retain is the User Settings link. Occasionally we need to direct a student to the User Settings to change from Mathjax to images rendering mode when they have problems with MathJax. I think there should still be some way to access the user settings in no-navigation mode, for instance a settings icon in the titlebar. A link to the Grades page might also be desirable, again perhaps as an icon in the title bar.
The other component of the left-hand pane that I think should be retained in some form is the assignment progress box:
![image](https://user-images.githubusercontent.com/8839416/106063259-fffacc80-614b-11eb-89c6-d190be5e157c.png)

**Comment by Alex-Jordan** [comment]
In User Settings, it seems like the only thing to be concerned about is the math display mode. For this use case, password and email address aren't relevant. Actually having those options there can confuse students. They can't generally change their password if their account was created through LTI.

Maybe something that allows you to select the math display mode from a popup menu could appear down where the checkbox for "Show correct answers" is (if it appears), before the submit buttons. That might be useful in general.

If the student navigates to the WW grades page, then from there they will have links to other assignments (assuming no modification happens to the Grades page). It would be a back door for navigating beyond the one problem set. I'd want to avoid that and it seems reasonable in this use case to expect students to use the LMS to look at grades.

The problems progress panel is one to think about. Locally, we are more interested in using this with GW quizzes, where that panel is not present. So even if it takes up space for homework, it wouldn't be a factor with quizzes.

**Reply by awmorp** [reply]
> select the math display mode from a popup menu

Good idea!

**Comment by somiaj** [comment]
It is now possible to disable students access pages such as the homework page, and so froth, making it so the main menu can be minimized. I think that addresses this discussion so closing.
